pkg/authenticator/authenticator.go

```diff
@@ -250,32 +285,31 @@ func (auth *Authenticator) Authenticate() ([]byte, error) {
 		return nil, log.RecordedError(log.CAKC027, err)
 	}
 
-	return DataResponse(resp)
+	err = validateResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	return readBody(resp)
```

Comment:
`readBody` is a private function in another file.
Just a thought...
I know it's valid `go` because they reside in the same package (which means the same folder in out convention), but is less readable IMO.
Maybe it should be moved to a different package? 🤔

Reply:
i think it makes sense to have it in the `requests` package

Reply:
Do you plan to move it to there?

Reply:
Do you plan to move it to there?

Reply:
it's already there :)

Reply:
created a PR for this - https://github.com/cyberark/conjur-authn-k8s-client/pull/187

we need to add UTs so it's not ready yet and i have other pressing matters. you can review the changes in your spare time, thanks!